index.js

```diff
@@ -74,85 +74,91 @@ function serverHandler(req, res) {
             //this is already a string when sent as JSON
             data = Buffer.concat(buffer, bufferLength);
         }
-        
-        // if a secret is configured, make sure the received signature is correct
-        if (self.secret) {
-            var signature = req.headers['x-hub-signature'];
-
-            if (!signature) {
-                self.logger.error('secret configured, but missing signature, returning 403');
-                return reply(403, res);
-            }
 
-            signature = signature.replace(/^sha1=/, '');
-            var digest = Crypto.createHmac('sha1', self.secret).update(data).digest('hex');
-
-            if (signature !== digest) {
-                self.logger.error('got invalid signature, returning 403');
+        self.getSecret(req, function (err, secret) {
+            if (err) {
+                this.logger.error('error getting secret for' +  res.url + 'returning 403');
```

Comment:
thanks, it went through a few iterations before everything came back into the same function.